src/components/Todo/EditionForm.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import Datepicker from "react-datepicker";
+
+export default class EditionForm extends React.Component {
+  state = {
+    editTodoName: this.props.todo.name,
+    editDueDate: this.props.todo.dueDate
+   }
+
+  render() {
+    const { onUpdate, todo } = this.props;
+    const { editTodoName, editDueDate } = this.state;
+    const canSubmit = editTodoName === "" || editDueDate === null;
+
+    return (
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          if (!canSubmit) return;
+          onUpdate(todo.id, {name: editTodoName, dueDate: editDueDate})
+      }}
+      >
+        <input
+          className="input is-primary is-small"
+          autoFocus
+          value={editTodoName}
+          onChange={e => this.setState({ editTodoName: e.target.value })}
+        />
+        <Datepicker
+          className="input is-primary is-small"
+          placeholderText="Click to select a date"
+          selected={this.state.editDueDate}
+          onChange={newDate => this.setState({ editDueDate: newDate })}
+        />
+        <button
+          type="submit"
+          className="button is-primary is-small"
+          disabled={ canSubmit }
```

Comment:
This line would NOT be correct if canSubmit was correct. The button should be disabled when you cannot submit so `!canSubmit`